plugins/connection/aws_ssm.py

```diff
@@ -304,10 +309,10 @@ def start_session(self):
             raise AnsibleError("failed to find the executable specified %s."
                                " Please verify if the executable exists and re-try." % executable)
 
-        profile_name = ''
+        profile_name = self.get_option('profile') or ''
```

Comment:
Should this be better set as an entry on the configuration blob above? 
Perhaps with AWS_PROFILE support too?

```
profile:
   ...
   default: ''
   env:
    - name: AWS_PROFILE
```